File: app/src/main/java/io/github/pshegger/infobook/fragments/BaseInfoBookFragment.java
package io.github.pshegger.infobook.fragments;

import io.github.pshegger.infobook.GWApiService;
import io.github.pshegger.infobook.InfoBookApplication;

/**
 * Created by pshegger on 2015.10.13..
 */
public abstract class BaseInfoBookFragment extends NamedFragment {
    protected GWApiService getAPIService() {
        return InfoBookApplication.getInstance().getService();
    }
}

File: app/src/main/kotlin/io/github/pshegger/infobook/fragments/BaseInfoBookFragment.kt
package io.github.pshegger.infobook.fragments

import io.github.pshegger.infobook.GWApiService
import io.github.pshegger.infobook.InfoBookApplication

/**
 * Created by pshegger on 2015. 11. 10..
 */
abstract class BaseInfoBookFragment : NamedFragment() {
    fun apiService(): GWApiService = InfoBookApplication.instance.service
}